draft-specifications/CapTP Specification.md

```diff
@@ -0,0 +1,978 @@
+connection at the same time and only one of these connection attempts must
+succeed. The way to check which attempt should succeed is to sort the serialized
+`session-pubkey`, the highest of the two wins.
+
+### Constructing and sending
+
+The `captp-version` MUST be `1.0`.
+
+The `session-pubkey` is the public key part of the per-session key pair
+generated for this connection. This is serialized in accordance with
+[Cryptography](#cryptography)
+
+The `acceptable-location` is a OCapN Locator which represents the location where
+the session is accessable.
+
+The `acceptable-location-sig` is the signature of the serialized
+`acceptable-location`. The signature is created using the private key from the
+per-session key pair. This is serialized in accordance with
+[Cryptography](#cryptography)
+
+### Receiving
+
+The `captp-version` MUST be equal to `1.0`. If the version does not match, the
+connection MUST be aborted.
+
+The `acceptable-location-sig` MUST be valid that the `session-pubkey` provided a
+valid signature of `acceptable-location`.
+
+## [`op:bootstrap`](#op-bootstrap)
+
+This operation requests the bootstrap object to be available at the specified
+`answer-position`. The message looks like:
+
+```text
+<op:bootstrap [answer-position    ; positive integer or 0
+               resolve-me-desc]>  ; desc:import-object | desc:import-promise
+```
+
+### The bootstrap Object
+
+The bootstrap object is responsible for providing access to local objects on the
+session. It has two different behaviors, these are selected using the
+conventional CapTP method mechanism of sending a symbol as the first argument,
+the following methods are available:
+
+-   `fetch`
+-   `deposit-gift`
+-   `withdraw-gift`
+
+#### `fetch` Method
+
+This method is used to fetch an object from the bootstrap object. To use it you
+need a `swiss-number` which is a Binary Data type. This swiss number should
+correspond an object which exists in this session. The result will be the object
+which corresponds to this `swiss-number` or an error if the object does not
+exist or a swiss number was not provided.
+
+An example of how to use this method is:
+
+```text
+<op:deliver <desc:export 0>          ; Remote bootstrap object
+            ['fetch                  ; Argument 1: Symbol "fetch"
+             swiss-number]           ; Argument 2: Binary Data
+            3                        ; Answer position
+            <desc:import-object 5>>  ; object exported by us at position 5 should provide the answer
+```
+
+#### `deposit-gift` Method
+
+The deposit gift method is used in conjunction with sending a [Third Party
+Handoff](#third-party-handoffs). This method is used to deposit a gift which has
+been sent to the bootstrap object. It has two arguments:
+
+1.  A gift ID that is positive integer.
+2.  A `desc:import-object` or `desc:import-promise` which has been exported
+    within the given CapTP session.
+
```

Comment:
missing an argument:
3. VatId of the recipiant of the gift.

Reply:
In Spritely Goblins case here it would refer to the publiv key (or the cryptographics hash there of) of the endpont that is to get the gift.

@markm has the nifty phrase "My name is how you know it is me." which captures the essence of the idea of self authenticated designators.

Reply:
I don't think this is needed or adds anything to the handoff process.

The `desc:handoff-give` and the `desc:handoff-receive` already contain enough information that ensures that only the exporter can pick up the gift left for them.